packages/components/table/src/TableCell/TableCell.tsx

```diff
@@ -67,7 +98,31 @@ function _TableCell(
       }}
       testId={testId}
     >
-      {children}
+      {isSortable ? (
+        <button
+          aria-label={`Sort ${
+            isSorted === TableCellSorting.ascending
+              ? TableCellSorting.descending
+              : TableCellSorting.ascending
+          } by ${children}`}
```

Comment:
what happens if `children` is not string and it is JSX? Won't this interpolation then show something like `by Function x()`?

Reply:
wanted to ask the same, I guess we need to check the children type first and use it if it's `string` only